### PR TITLE
feat(credentialOffer): issuance flow state and interaction helpers

### PR DESCRIPTION
**Pull request by mnzaki**

### CODE REVIEW

File: src/lib/interactionManager/interaction.ts [hunk 1]
```diff
@@ -163,6 +165,50 @@ export class Interaction {
     )
   }
 
+  public async issueSelectedCredentials(
+    offerMap?: {
+      [k: string]: (inp?: any) => Promise<{ claim: any, metadata?: any, subject?: string }>
```
Comment (clauxx) [inline src/lib/interactionManager/interaction.ts:170]: Why is this map of promises and not a map of objects?

Comment (clauxx) [reply src/lib/interactionManager/interaction.ts:170]: Also why is it optional?

File: src/lib/interactionManager/interaction.ts [hunk 2]
```diff
@@ -56,6 +56,8 @@ export class Interaction {
     responder?: Identity
   }
 
+  counterparty?: Identity
```
Comment (clauxx) [inline src/lib/interactionManager/interaction.ts:61]: Why is this necessary? I thought this is covered by the `participants` member.

File: src/lib/interactionManager/interaction.ts (hunk 1 above)
Comment (mnzaki) [reply src/lib/interactionManager/interaction.ts:170]: Each handler returns a promise to allow for doing async things during the actual issuance process, in case the app logic requires it. This is so that we let this happen last second instead of requiring the app (backend) to already do all the work before even knowing what the "offer selection" is :D

File: src/lib/interactionManager/interaction.ts (hunk 2 above)
Comment (mnzaki) [reply src/lib/interactionManager/interaction.ts:61]: it is not covered, it can be either the value of `requester` or `responder`, depending on what "our" role is

File: src/lib/interactionManager/interaction.ts (hunk 1 above)
Comment (mnzaki) [reply src/lib/interactionManager/interaction.ts:170]: It is optional to allow for the "simple" use case of just empty claims

File: src/lib/interactionManager/interaction.ts (hunk 2 above)
Comment (clauxx) [reply src/lib/interactionManager/interaction.ts:61]: It's always the `responder` in this case, but I get your point. In this case, I think the `getSummary` method should return the `counterparty`'s summary instead of the `requester`'s.
